src/node/interfaces.cpp

```diff
@@ -621,8 +651,13 @@ class ChainImpl : public Chain
     }
     bool isReadyToBroadcast() override { return !::fImporting && !::fReindex && !isInitialBlockDownload(); }
     bool isInitialBlockDownload() override {
-        assert(std::addressof(::ChainstateActive()) == std::addressof(m_node.chainman->ActiveChainstate()));
-        return m_node.chainman->ActiveChainstate().IsInitialBlockDownload();
+        const CChainState* active_chainstate;
+        if (m_node.chainman) {
```

Comment:
does it make sense to run a node without a chainman? Imo it might make more sense to use Assert, like it is already done in `waitForNotificationsIfTipChanged`.

Maybe the assert can even be wrapped into a private member:

```
private:
 ChainstateManager& chainman() { return *Assert(m_node.chainman); }